rfc/system/2704-release-cycle-process-updates.md

```diff
@@ -0,0 +1,107 @@
+
+Not strictly forcing a time-constraint on the Flyte release cycle means that if a substantial number of changes is merged, perhaps due to a security issue or just a rapid pace of feature development, we can always bring up the timeline of the release.
+
+### Components Included in Version Alignment
+Not all of the Flyte repos will follow this naming semantic. We will only align these:
+
+* Propeller
+* Admin
+* Console
+* datacatalog
+* flytectl
+* flytesnacks
+* Flytekit
+* flytekit-java
+
+The last two we are going to tie together for now, but realize that we may want to unpin in the future.
+
+The above are the more user facing repos. The components that will not be tied to this scheme are:
+* flyteidl
+* flytestdlib
+* flyteplugins
+* flytecopilot
+
+
+#### Helm Charts
+The Helm charts should also follow the Flyte release version. Unlike the individual components however, the Helm version and the Flyte version should be identical down to the patch version. While Helm charts do have two separate versions, one for the chart and one for the app, according to their [guidelines](https://codefresh.io/docs/docs/new-helm/helm-best-practices/), "These are unrelated and can be bumped up in any manner that you see fit. You can sync them together or have them increase independently."
+
+
+### Release Branches and Patching
+After each minor release, a release branch will be created. There will be no alignment of patch versions across the components. That is, By the end of the `1.3.x` release cycle, `flyteadmin` may be on `1.3.8` and `flytepropeller` may be on `1.3.2`.
+
+#### Patching older releases
+When developing bug fixes, by default we will continue to develop off of master, which will not be the stable branch. After such bug fixes are merged, it will be the responsibility of the developer to ensure that the patches are also applied to prior releases. At the current time, we propose only supporting one release back. That is, if `flytepropeller` has a bug fix that results in `v1.3.0b0` that patch will be applied to the `v1.2.x` release, but not the `v1.1.x` release.
+
+We also propose that beta patch versions be merged into the release branch when patching prior releases. For example, assuming no patches have yet to be made to the `v1.2.0` release, when porting a bug fix that resulted in `v1.3.0b0` onto the `release-v1.2` branch, the developer can first release `v1.2.1b0` for testing into `release-v1.2` before releasing the `v1.2.1` release (also to `release-v1.2` of course). Such beta releases should be made (always, when it's a big change, at the discretion of the developer????).
+
+#### Why not develop on release branches?
+A proposal was made that:
+* Bug fixes should be made to the latest release branch when the big was discovered.
+* Features should be developed against master.
+
+We decided that this approach was less desirable because:
```

Comment:
How was this decided -- where did the discussion happen [ I'd like to read those archives, if available ]?